boot/boot_serial/src/boot_serial_encryption.c

```diff
@@ -0,0 +1,319 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Copyright (c) 2020-2023 Nordic Semiconductor ASA
+ * Copyright (c) 2020 Arm Limited
+ */
+
+#include <assert.h>
+#include "bootutil/image.h"
+#include <../src/bootutil_priv.h>
+#include "bootutil/bootutil_log.h"
+#include "bootutil/bootutil_public.h"
+#include "bootutil/fault_injection_hardening.h"
+#include "bootutil/enc_key.h"
+
+#include "mcuboot_config/mcuboot_config.h"
+
+#ifdef MCUBOOT_ENC_IMAGES
+
+BOOT_LOG_MODULE_DECLARE(serial_encryption);
+
+/* Variables passed outside of unit via poiters. */
+static const struct flash_area *_fa_p;
```

Comment:
Does this still need to be a static. The only use I see here outside of a single function is ifdef'd out code. The statics are a problem for unit testing, which we aren't currently seeing because we aren't unit testing the serial code.